geode-log4j/src/main/java/org/apache/geode/logging/log4j/internal/impl/LogWriterAppender.java

```diff
@@ -302,10 +302,15 @@ public synchronized void startSession() {
 
   @Override
   public synchronized void stopSession() {
+    if (logWriter == null) {
+      pause();
+      return;
+    }
```

Comment:
I think you should just remove the early-out. In theory, an early-out or an assert could be added other places as well such as startSession() but it's better to be consistent in this case. The usage of LogWriterAppender is driven by the LoggingSession code and nothing will ever call startSession or stopSession without calling createSession. If it was a user-facing User API then I might be inclined to protected these method calls from a null logWriter. Even if we did add these protect clauses then I would think that we should ensure that all log statements (such as line 309) print out.